### ISSUE TITLE
Could you provide some training guidance?

### ISSUE THREAD

**Issue by maryawood** [google/deeppolisher#10]
Hello, I'm hoping to train a custom model using the `model_train_custom_loop.py` script provided in the repository, but I'm a bit uncertain about how to use it and have run into some hurdles attempting to figure it out on my own. A few questions I have:

- Should you run `make_images` for training and evaluation sets prior to using the script? If so, do you have guidance for how to generate the truth-to-ref alignment and the high confidence region BED file?
- How should I generate the parameters configuration file needed as input to the script?
- Is it equally acceptable to train with a haploid or diploid example?
- Can you include more than one individual in training and/or evaluation data sets? If so, is there a preferred way to structure the data to support this?

Thanks!

**Comment by maryawood** [comment]
Just wanted to ping again on this since it has been a couple of weeks, and I'm still experiencing challenges with using the script.

**Comment by kishwarshafin** [comment]
Hi @maryawood ,

Apologies for being late on this. I have turned on email notifications for this repo so if future it would not get dropped. Currently, we do not have a public doc describing the training pipeline. 

> Should you run make_images for training and evaluation sets prior to using the script? If so, do you have guidance for how to generate the truth-to-ref alignment and the high confidence region BED file?

This is the way we generate training:

```bash
polisher make_images \
--bam $BAM \
--fasta $FASTA \
--truth_to_ref $TRUTH_TO_REF \
---region_bed $REGION_BED \
---output $OUTPUT \
---cpus $CPUS \
---training_mode
```

The region bed is the high confidence bed. However, then we shuffle the examples like deepconsensus does and then train the model. If you have data that you can share and want us to train a model then feel free to reach out.

> How should I generate the parameters configuration file needed as input to the script?

Yes, the make_images will generate the pbtxt files, you have to point to them. You need to split between training examples and tuning examples.

> Is it equally acceptable to train with a haploid or diploid example?

The model expects the underlying data to be haploid. It will only give you one VCF so diploid assumptions would not work.

> Can you include more than one individual in training and/or evaluation data sets? If so, is there a preferred way to structure the data to support this?

You can, you can generate all of the data in the output directory and shuffle would concat them together during that step.

I recommend you look at this: https://github.com/google/deepconsensus/tree/r1.2/docs as DeepPolisher is mostly derived from DeepConsensus so it will be helpful in many instances.

**Comment by maryawood** [comment]
Hi, thank you so much for your response!

Could you explain the expected format for the truth_to_ref input? Is this just an end-to-end alignment of the "ground truth" reference to the polished reference? Do you have any recommendations for methodology when creating it?

I still need to take a closer look at the DeepConsensus documentation, but it looks like it is pointing to a python file for configuration rather than a pbtxt file. I only got tensorflow record files and JSON file out of my attempt at running `make_images` in training mode. Could you tell me a bit more about what is needed for custom configuration?

**Comment by kishwarshafin** [comment]
> Could you explain the expected format for the truth_to_ref input? Is this just an end-to-end alignment of the "ground truth" reference to the polished reference? Do you have any recommendations for methodology when creating it?

we use dipcall to align each truth (in our case it's Q100 assembly) to the raw assembly using dipcall. The output bam is then used as `truth_to_ref` input.

> I still need to take a closer look at the DeepConsensus documentation, but it looks like it is pointing to a python file for configuration rather than a pbtxt file. I only got tensorflow record files and JSON file out of my attempt at running make_images in training mode. Could you tell me a bit more about what is needed for custom configuration?

You need to update the python file with the pbtxt locations so it can read from there. So you need to write this function in:

https://github.com/google/deeppolisher/blob/r0.1.0/polisher/models/model_configs.py

```
def _set_data_hparams_for_haploid(params):
  params.train_path = '/path/to/train/*.tfrecords.gz'
  params.eval_path = '/path/to/tune/*.tfrecords.gz'
  params.test_path = '/path/to/test/*.tfrecords.gz'
  params.n_examples_train = NUM_EXAMPLES_TRAIN # Can get from pbtxt
  params.n_examples_eval = NUM_EXAMPLES_TUNE
  params.n_examples_test = NUM_EXAMPLES_TEST
  params.max_passes = encoding.get_max_coverage()

```

**Comment by maryawood** [comment]
Thank you for your help! I've been able to successfully run training using this info
